nautobot_golden_config/api/serializers.py

```diff
@@ -68,13 +69,31 @@ class GoldenConfigSettingSerializer(TaggedObjectSerializer, CustomFieldModelSeri
     url = serializers.HyperlinkedIdentityField(
         view_name="plugins-api:nautobot_golden_config-api:goldenconfigsetting-detail"
     )
+    scope = serializers.JSONField(required=False)
+    dynamic_group = NestedDynamicGroupSerializer(required=False)
 
     class Meta:
         """Set Meta Data for GoldenConfigSetting, will serialize all fields."""
 
         model = models.GoldenConfigSetting
         fields = "__all__"
 
+    def validate(self, data):
+        """Validate scope & dynamic_group are not both submitted."""
+        if data.get("scope") and data.get("dynamic_group"):
+            raise serializers.ValidationError("Payload can only contain `scope` or `dynamic_group`, but not both.")
+        return data
+
+    def create(self, validated_data):
+        """Overload to handle ability to post scope instead of dynamic_group."""
```

Comment:
Can you add a comment about how/why this works? It took me a few min to remember that the scope setter will actually set dynamic group, and trying to be a bit more kind to someone in the future.

Reply:
Done